src/GlobalHelper.php

```diff
@@ -129,7 +130,28 @@ public static function getCache(): CacheItemPoolInterface&CacheInterface
     public static function setupDefaultCache(): void
     {
         if (!isset(self::$cache)) {
-            self::setCache(new FilesystemAdapter(directory: sys_get_temp_dir() . '/castor'));
+            self::setCache(new FilesystemAdapter(directory: self::getGlobalDirectory() . '/cache'));
         }
     }
+
+    public static function setHomeDirectory(string $homeDirectory): void
+    {
+        self::$homeDirectory = $homeDirectory;
+    }
+
+    /**
+     * @throws \RuntimeException If the user home could not reliably be determined
+     */
+    public static function getHomeDirectory(): string
+    {
+        return self::$homeDirectory ??= PlatformUtil::getUserDirectory();
+    }
+
+    /**
+     * @throws \RuntimeException If the user home could not reliably be determined
+     */
+    public static function getGlobalDirectory(): string
+    {
+        return self::getHomeDirectory() . '/.castor';
+    }
```

Comment:
I would 

1. remove `getHomeDirectory`, it's not used
2. rename `setHomeDirectory` in `getGlobalDirectory`